### PR TITLE
Fix: Fixed issue where path name wasn't visible when grouping libraries by path

### LINKED ISSUE #8144: Bug: Group by path in Libraries is missing path name
### Description

It seems to be a problem for the visualisation of the path in libraries when the visualisation is "group by path"

Cf screenshot.

Sorry for my english from France ;-)

### Steps To Reproduce

Add local path and unc path to a library

Group by path for the visualisation of the library

### Expected behavior

The last folder level should appear in front of the path

### Files Version

2.1.5.0

### Windows Version

Édition	Windows 11 Professionnel Version	21H2 Installé le	‎02/‎11/‎2021 Build du système d’exploitation	22000.469 Expérience	Pack d’expérience de fonctionnalités Windows 1000.22000.469.0

### Relevant Assets/Logs

image

### PR DESCRIPTION
**Pull request by ashrafmansuri**
**Resolved / Related Issues**
- [ ] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #8144...

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [ ] Are there any other steps that were used to validate these changes?
   1. Open app ...
   2. Click settings button ...

**Screenshots (optional)**
Add screenshots here.

### CODE REVIEW

**Review by ashrafmansuri — COMMENTED** [review]
Made changes in GroupingHelper added method to check if it's a UNC path. Reverted changes in GroupedHeaderViewModel made earilier.

File: src/Files.App/Utils/Storage/Collection/GroupingHelper.cs [hunk 1]
```diff
@@ -120,7 +120,7 @@ public static (Action<GroupedCollection<ListedItem>>, Action<GroupedCollection<L
 					var model = x.Model;
 					model.ShowCountTextBelow = true;
 					var parentPath = PathNormalization.GetParentDir(first.ItemPath.TrimPath());
-					model.Text = System.IO.Path.GetFileName(parentPath);
+					model.Text = IsUNCPath(parentPath) ? GetUNCFileName(parentPath) : System.IO.Path.GetFileName(parentPath);
```
Comment (hishitetsu) [inline src/Files.App/Utils/Storage/Collection/GroupingHelper.cs:123]: This is not just a UNC problem, but a problem when the parent is not a folder. What if the library location is the route of a drive?

Comment (ashrafmansuri) [reply src/Files.App/Utils/Storage/Collection/GroupingHelper.cs:123]: Yeah I guess so. How do I go about it?

Comment (hishitetsu) [reply src/Files.App/Utils/Storage/Collection/GroupingHelper.cs:123]: You can check `System.IO.Path.GetFileName()` first. If `System.IO.Path.GetFileName()` is null or empty, then you can try `GetUNCFileName()`. The method name should be changed though.

Comment (ashrafmansuri) [reply src/Files.App/Utils/Storage/Collection/GroupingHelper.cs:123]: For all the System.IO.Path.GetFileName() in the project?

Comment (hishitetsu) [reply src/Files.App/Utils/Storage/Collection/GroupingHelper.cs:123]: Just here is fine.

Comment (ashrafmansuri) [reply src/Files.App/Utils/Storage/Collection/GroupingHelper.cs:123]: Okay

Comment (ashrafmansuri) [reply src/Files.App/Utils/Storage/Collection/GroupingHelper.cs]: Resolved Please check

**Review by ashrafmansuri — COMMENTED** [review]
Check with Drives root folder and a Network Path.

File: src/Files.App/Utils/Storage/Collection/GroupingHelper.cs [hunk 2]
```diff
@@ -165,5 +165,12 @@ public static string GetGroupSizeKey(long size)
 			(1000000, "ItemSizeText_Medium".GetLocalizedResource(), "1 MiB".ConvertSizeAbbreviation()),
 			(16000, "ItemSizeText_Small".GetLocalizedResource(), "16 KiB".ConvertSizeAbbreviation()),
 		};
+
+		public static string GetFolderName(string path)
```
Comment (hishitetsu) [inline src/Files.App/Utils/Storage/Collection/GroupingHelper.cs:169]:
```suggestion
		private static string GetFolderName(string path)
```

**Review by hishitetsu — APPROVED** [review]
LGTM except for one minor point.

**Comment by ashrafmansuri** [comment]
> LGTM except for one minor point.

Committed.

**Comment by yaira2** [comment]
@ashrafmansuri thank you for contributing to Files.

**Comment by ashrafmansuri** [comment]
> @ashrafmansuri thank you for contributing to Files.

You're welcome. 😄

**Pull request merged**